genie-web/src/test/java/com/netflix/genie/web/jpa/services/JpaJobPersistenceServiceImplUnitTests.java

```diff
@@ -740,6 +741,49 @@ public void unresolvedJobReturnsEmptyJobSpecificationOptional() {
         Assert.assertFalse(this.jobPersistenceService.getJobSpecification(UUID.randomUUID().toString()).isPresent());
     }
 
+    /**
+     * If a job isn't found on querying for v4 flag.
+     */
+    @Test(expected = GenieJobNotFoundException.class)
+    public void noJobUnableToGetV4() {
+        Mockito
+            .when(this.jobRepository.findByUniqueId(Mockito.anyString(), Mockito.eq(IsV4JobProjection.class)))
+            .thenReturn(Optional.empty());
+
+        this.jobPersistenceService.isV4(UUID.randomUUID().toString());
+    }
+
+
+    /**
+     * If v4 is false in db then return false.
+     */
+    @Test
+    public void v4JobFalseReturnsFalse() {
+        final JobEntity jobEntity = Mockito.mock(JobEntity.class);
+        Mockito
+            .when(this.jobRepository.findByUniqueId(Mockito.anyString(), Mockito.eq(IsV4JobProjection.class)))
+            .thenReturn(Optional.of(jobEntity));
+
+        Mockito.when(jobEntity.isV4()).thenReturn(false);
+
+        Assert.assertFalse(this.jobPersistenceService.isV4(UUID.randomUUID().toString()));
+    }
+
+    /**
+     * If v4 is true in db then return true.
+     */
+    @Test
+    public void v4JobTrueReturnsTrue() {
+        final JobEntity jobEntity = Mockito.mock(JobEntity.class);
+        Mockito
+            .when(this.jobRepository.findByUniqueId(Mockito.anyString(), Mockito.eq(IsV4JobProjection.class)))
+            .thenReturn(Optional.of(jobEntity));
+
+        Mockito.when(jobEntity.isV4()).thenReturn(true);
+
+        Assert.assertTrue(this.jobPersistenceService.isV4(UUID.randomUUID().toString()));
+    }
+
```

Comment:
Add a test for PersistenceService throwing exception?

Reply:
Did that

Reply:
Still is valid. The test was missing